### PR TITLE
LPS-153058 | Bug Test Automation | Master

### PR DESCRIPTION
**Pull request by rafaellajordao**
# Test Automation for LPS-109991
### When a variant is edited, other variants lose modifications

Hi! :)
This pr is corresponding to this ticket: [LPS-153058](https://issues.liferay.com/browse/LPS-153058)

### CODE REVIEW

**Comment by rafaellajordao** [comment]
ci:test:sf

**Comment by rafaellajordao** [comment]
ci:test:tango

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTesting.testcase [hunk 1]
```diff
@@ -635,6 +635,84 @@ definition {
...
+
+			ABTest.createABTest(
+				goal = "Bounce Rate",
+				testTitle = "AB test title");
+
+			ABTest.createVariant(variantName = "Variant Name 1");
+		}
+
+		task ("Edit variant page and add some fragment to the page") {
+			Click(
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTesting.testcase:654]: this click should be in a macro, something like "ABTest.editVariant(ABTest.createVariant(variantName = "Variant Name 1")

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/abtest/ABTest.path [hunk 2]
```diff
@@ -85,6 +85,11 @@
 	<td>//button[contains(@class,'btn-outline-secondary') and *[name()='svg'][contains(@class,'lexicon-icon-pencil')]]</td>
 	<td></td>
 </tr>
+<tr>
+	<td>PENCIL_BUTTON_BY_NAME</td>
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/abtest/ABTest.path:89]: I would change the name of this path to EDIT_VARIANT_BUTTON

**Review by CarlosBrichete — CHANGES_REQUESTED** [review]

**Comment by rafaellajordao** [comment]
ci:test:tango

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTesting.testcase [hunk 3]
```diff
@@ -635,6 +635,76 @@ definition {
...
+		task ("Create second variant and add some fragment to the page") {
+			ABTest.createVariant(variantName = "Variant Name 2");
+
+			ABTest.editVariant(variantName = "Variant Name 2");
+
+			PageEditor.addFragment(
+				collectionName = "Basic Components",
+				fragmentName = "Heading");
+
+			Click(
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTesting.testcase:674]: This also should be in a macro called saveVariant()

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTesting.testcase [hunk 4]
```diff
@@ -635,6 +635,76 @@ definition {
...
+		}
+
+		task ("Edit variant page and add some fragment to the page") {
+			ABTest.editVariant(variantName = "Variant Name 1");
+
+			PageEditor.addFragment(
+				collectionName = "Basic Components",
+				fragmentName = "Button");
+
+			Click(
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTesting.testcase:660]: This also should be in a macro called saveVariant()

**Review by CarlosBrichete — CHANGES_REQUESTED** [review]

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTesting.testcase (hunk 4 above)
Comment (CarlosBrichete) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTesting.testcase:660]: Use ABTest. saveVariant here

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTesting.testcase (hunk 3 above)
Comment (CarlosBrichete) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTesting.testcase:674]: Use ABTest. saveVariant here

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTesting.testcase [hunk 5]
```diff
@@ -635,6 +635,76 @@ definition {
...
+			ContentPages.viewFragmentText(
+				fragmentName = "button",
+				id = "link",
+				text = "Go Somewhere");
+
+			PageEditor.removeFragment(fragmentName = "Button");
+
+			PageEditor.viewFragmentNotPresent(fragmentName = "Button");
+
+			Click(
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/ABTesting.testcase:693]: Use ABTest. saveVariant here

**Review by CarlosBrichete — CHANGES_REQUESTED** [review]

**Comment by rafaellajordao** [comment]
ci:test:sf

**Comment by rafaellajordao** [comment]
ci:test:tango

**Review by CarlosBrichete — APPROVED** [review]

**Comment by CarlosBrichete** [comment]
LGTM! :)

**Comment by rafaellajordao** [comment]
ci:forward

**Pull request closed without merging**